frontend/src/components/VirtualList/VirtualItem.tsx

```diff
@@ -58,7 +58,7 @@ export default class VirtualItem extends React.Component<VirtualItemProps, Virtu
     const { style, className, item } = this.props;
     const key = 'VirtualItem_' + ('namespace' in item ? 'Ns' + item.namespace + '_' + item.name : item.name);
     return (
-      <tr style={style} className={className} role="row" key={key}>
+      <tr style={style} className={className} role="row" key={key} data-cy={key}>
```

Comment:
"data-key" instead? Just trying to add "data-*" attributes more domain-specific, no need to use a "cypress" prefix for them, just annotate things that may identify the value.

https://github.com/kiali/kiali/commit/cd5cbffe1cfd9fa897a9d4a20ed584bfd3048b37#diff-9cd721a0ae66bcf3e0135496c5d43d07de1da7ae58eec132049b012b60031817R59

Reply:
The [recommendation](https://docs.cypress.io/guides/references/best-practices#Selecting-Elements) seems to be to have the key be static/constant and just have the value change. Seems to be common to include a custom command around this like:
```
Cypress.Commands.add('getBySel', (selector, ...args) => {
  return cy.get(`[data-test=${selector}]`, ...args)
})

cy.getBySel('value-of-data-test')
```
We should decide on what key we're using and just go with that. `data-cy` is fine with me.

Reply:
I read that recommendation that the best practice is to provide a good context.
If there should be only one data- attribute, it can be anything, but I see that we need to add data- attributes in places where there are existing "aria-" or  "data-" ones and also we have patterns where the same data- may have different values.
In example, you may want to add a "namespace" and a "workload" name in some pattern.

So I encourage to use data attributes with context, so, if you are going to add a "namespace", better to use a "data-namespace" or "data-workload", there would be cases where you may have multiple lists/tables per namespace, rendering workloads/apps. I see it confusing to use the same "data-cy" where in some cases the value is a namespace and in others a workload.

I agree that if it's something new and there is no clear attribute, the author can define the "data-" naming that better suits.

Reply:
> So I encourage to use data attributes with context, so, if you are going to add a "namespace", better to use a "data-namespace" or "data-workload", there would be cases where you may have multiple lists/tables per namespace, rendering workloads/apps. I see it confusing to use the same "data-cy" where in some cases the value is a namespace and in others a workload.

The context goes in the value e.g. `data-cy="workload-details-logs-tab"` which is consistent with other html tags like `class="expanded"` or `aria-label="filter_select_type"`. This is how the cypress real world app is setup: https://github.com/cypress-io/cypress-realworld-app/search?q=data-test. It also seems to be the convention that the openshift console uses: https://github.com/openshift/console/search?q=data-test. The "Selector Playground" optimizes for using `data-cy` or `data-test`. It's nice to be able to quickly search for all "special" testing elements using a consistent attribute and it lets you add some helper functions like the command above. Can we start with `data-cy` or `data-test` and, if need be, add additional `data-test-*` attributes later?

Reply:
I'd prefer that if we have a "data-namespace", we keep using the same when you need to add a "data-namespace".
What I don't see good is to use a data-cy={namespace} in one place (a <ul> element) and a data-cy={workload} in an inner one. It's better to use a data-namespace={bookinfo} or a data-workload={a-client}, then after some months, everyone should understand what type of value needs that attribute.

Or at least, I suggest to use a data-test-namespace={'bookinfo'}, data-test-workload={'a-workload'], etc

Please, let's unify conventions now that we are starting, I don't mind updating the tests I wrote, but let's not have different ways of naming depending on the author, otherwise, we will outdated these tests as soon as some fail as we simply update the value to comply (as it happened with some of the existing JEST tests the project have).

Reply:
Or we can use a "data-cy" when there is only one attribute and use "data-cy-*" when there is more than one close to the same area (I'm thinking in a ul -> li, in cases like that I'd prefer to have a data-cy-namespace/data-cy-workload).

Reply:
> What I don't see good is to use a data-cy="namespace" in one place (a "ul" element) and a data-cy="workload" in an inner one. It's better to use a data-namespace="bookinfo" or a data-workload="a-client", then after some months, everyone should understand what type of value needs that attribute.

Agreed just having the `namespace` as the value probably doesn't give you enough context and can easily conflict with other elements that may share a `data-cy=namespace`. You can provide additional context in the value however like `data-cy=list-item-${item.type}-${item.namespace}`. Here's a similar example from the cypress demo app: https://github.com/cypress-io/cypress-realworld-app/blob/develop/src/components/UserListItem.tsx#L14.

Having a `data-namespace` attribute suffers from a similar lack of context. If you've got multiple elements on the screen with `data-namespace` attributes, then it's ambiguous which one you select with `cy.get('[data-namespace="bookinfo"]')`. It could also lead to a proliferation of attribute types: `data-workload`, `data-service`, `data-app`, `data-virtualservice`. To me, it's more unclear when you use/create a new attribute type vs just using `data-cy` or `data-test`.

> Please, let's unify conventions now that we are starting

Yes! Let's do this. Ultimately we can choose whatever convention works best for us. I would just prefer we start with the recommended way and then as we learn more and get more experience with cypress we can deviate from that. I'd also prefer we use `data-test` or `data-test-*` over `data-cy`. `cy` has too much overlap with cytoscape and `data-test` is more understandable that it's an attribute for testing purposes only.

Reply:
I like to use the `data-test` as a general rule and `data-test-*` if there are cases that may require more.
About the concatenating in the string the context, it can be a start, but I guess it will depend on a case by case.

From the cypress examples, we can get some tips but we can adapt, the project is mature and this is more thinking about what would be easier to maintain in several months/years (we have some lessons learned from the JEST tests and that's why I wanted to discuss this even if "naming" may be over-discussed).

I'll start from my side reviewing the tests I did for the overview page and I'll adapt them to use the "data-test" / "data-test-*" to move into that direction.

Reply:
I also changed this PR to use `data-test`